# Libraries Import

In [5]:
import matplotlib.pyplot as plt
plt.style.use("seaborn")
# import MetaTrader5 as mt5
import plotly.express as px
from prepare_data import *
from Strategies.Stratey_bb_adx_rsi_ema200_1MIN import Strategy_bb_adx_rsi_ema200_1MIN
from Strategies.Strategy_rsi_stoch_ema50 import Strategy_rsi_stoch_ema50_1H

# Import Data

In [6]:
data = import_csv_data("./csv_files/15MIN/US30_15MIN.csv")
data = clean_data(data)


# Prepare Data

In [7]:
data = prepare_data(data)

e:\Programmes\python\lib\site-packages\ta\trend.py:769: RuntimeWarning:

invalid value encountered in double_scalars

e:\Programmes\python\lib\site-packages\ta\trend.py:774: RuntimeWarning:

invalid value encountered in double_scalars



# Parameters

In [23]:
ratio = 1.4
security_sl = 71
security_sl_1 = security_sl
security_sl_2 = security_sl
volume_control = 0
starting_balance = 1000
volume = 1
spread = 1*volume

# Run the Strategy

In [24]:
rsi_stoch_strategy = Strategy_rsi_stoch_ema50_1H(data, starting_balance=starting_balance, volume=volume, security_sl=security_sl, ratio=ratio, spread=spread)
result = rsi_stoch_strategy.run()

### Plot result

In [11]:
plot_trades_result(data, result)

In [25]:
px.line(result, x='close_datetime', y='pnl')

### % win vs % loss

In [13]:
loss = 0
win = 0
for index, row in result.iterrows():
  if row["profit"] < 0:
    loss = loss + 1
  else :
    win = win + 1

Win_ = (win) / (win + loss) * 100
Loss_ = loss / (win+loss) * 100


In [14]:
print(f"pourcentage Win {Win_}%")

pourcentage Win 43.02325581395349%


In [15]:
print(f"pourcentage Loss {Loss_}%")

pourcentage Loss 56.97674418604651%


In [26]:
max_drawdown(result["pnl"])

-0.33411175792788295

# Optimization

### Optimization SL

In [22]:
starting_balance = 1000
volume = 1
spread = 1*volume
ratio = 1.4

max_profit = {"security_sl": 0, "pnl": 0}

for i in range(300):
  rsi_stoch_strategy = Strategy_rsi_stoch_ema50_1H(data, starting_balance=starting_balance, volume=volume, security_sl=i, ratio=ratio, spread=spread)
  result = rsi_stoch_strategy.run()
  current_max_profit = result["pnl"].iat[-2]
  if current_max_profit > max_profit["pnl"]:
    max_profit["security_sl"] = i
    max_profit["pnl"] = current_max_profit

max_profit


{'security_sl': 71, 'pnl': 4479.020450426062}

# Optimization +++

#### Optimization SL ++

In [4]:
starting_balance = 1000
volume = 1
spread = 1*volume

max_profit = {"security_sl": 0, "ratio": 0, "pnl": 0}

for i in range(100, 300):
  ratio = i/100
  for j in range(100):
    rsi_stoch_strategy = Strategy_rsi_stoch_ema50_1H(data, starting_balance=starting_balance, volume=volume, ratio=ratio, spread=spread, security_sl=j)
    result = rsi_stoch_strategy.run()
    current_max_pnl = result["pnl"].iat[-2]
    if current_max_pnl > max_profit["pnl"]:
      max_profit["security_sl"] = j
      max_profit["ratio"] = ratio
      max_profit["pnl"] = current_max_pnl

max_profit
    

max_profit

{'security_sl': 34, 'ratio': 1.94, 'pnl': 5520.865020948026}

# Excel File

In [ ]:
result.drop(result.tail(1).index,inplace=True)
write_excel_file(result)

# Saved Results

### US30-1H (à faire)
#### Double sl : 1) ? < atr < ?, ratio=?
{'security_sl_1': ?, 'security_sl_2': ?, 'pnl': ?}

### US30-15MIN
#### Simple sl : sl = 34, ratio = 1.91
{'security_sl': 34, 'ratio': 1.94, 'pnl': 5520}

# Optimization Max Drawdown

In [ ]:
starting_balance = 1000
volume = 1
spread = 1*volume
ratio = 1.5

max_profit = {"security_sl": 0, "pnl": 0, "mdd": -100000}

for i in range(300):
  rsi_stoch_strategy = Strategy_rsi_stoch_ema50_1H(data, starting_balance=starting_balance, volume=volume, security_sl=i, ratio=ratio, spread=spread)
  result = rsi_stoch_strategy.run()
  current_max_dd = max_drawdown(result["pnl"])
  current_max_profit = result["pnl"].iat[-1]
  if current_max_dd > max_profit["mdd"]:
    max_profit["security_sl"] = i
    max_profit["pnl"] = current_max_profit
    max_profit['mdd'] = current_max_dd

max_profit